src/sentry/testutils/cases.py

```diff
@@ -212,21 +211,35 @@ def load_fixture(self, filepath):
             return fp.read()
 
     def _pre_setup(self):
+        pass
+
+    def _post_teardown(self):
+        pass
+
+    @pytest.fixture(autouse=True)
+    def pre_setup_and_post_teardown(self):
+        """
+        Massive hack to make py.test --pdb work correctly. When pytest is
+        passed this flag, it refuses to call _post_teardown() after any failing
+        or xfailing test, causing leaking teststate to crash subsequent tests
+        as well.
+
+        To fix this issue we make sure _pre_setup and _post_teardown are noops,
+        and drive the underlying code from Django using this fixture, which
+        seems to be correctly set up and torn down by pytest.
+
+        In the end the amount of times _pre_setup and _post_teardown are called
+        for regular tests does not change, it's always before/after each test
+        function.
+        """
         super(BaseTestCase, self)._pre_setup()
 
         cache.clear()
         ProjectOption.objects.clear_local_cache()
         GroupMeta.objects.clear_local_cache()
-
-    def _post_teardown(self):
+        yield
```

Comment:
Ahh yes, the bare pytest yield. I forget what this does but it was necessary for [this](https://develop.sentry.dev/testing/#inspecting-sql-queries-in-tests).

Reply:
Thanks for this link, I have stuff to add

Reply:
oh, `yield` "waits" for the test. stuff before yield is for setup and the stuff after is for teardown

Reply:
Ahh, yeah, that rings a bell in my head.

Reply:
https://github.com/getsentry/develop/pull/83

Reply:
Just realized Django has a SimpleTestCase, it might be better from a pure readability perspective to use that instead? See discussion: https://github.com/getsentry/develop/pull/83#discussion_r453995375